File: decompiled/sources/freemarker/template/TemplateNodeModel.java
package freemarker.template;

public interface TemplateNodeModel extends TemplateModel {
    TemplateSequenceModel getChildNodes() throws TemplateModelException;

    String getNodeName() throws TemplateModelException;

    String getNodeNamespace() throws TemplateModelException;

    String getNodeType() throws TemplateModelException;

    TemplateNodeModel getParentNode() throws TemplateModelException;
}

File: decompiled/sources/freemarker/template/TemplateNodeModel.kt
package freemarker.template

interface TemplateNodeModel : TemplateModel {
    @Throws(TemplateModelException::class)
    fun getChildNodes(): TemplateSequenceModel?

    @Throws(TemplateModelException::class)
    fun getNodeName(): String

    @Throws(TemplateModelException::class)
    fun getNodeNamespace(): String?

    @Throws(TemplateModelException::class)
    fun getNodeType(): String

    @Throws(TemplateModelException::class)
    fun getParentNode(): TemplateNodeModel?
}
